core/src/main/scala/kafka/server/ReplicaFetcherManager.scala

```diff
@@ -28,7 +28,6 @@ class ReplicaFetcherManager(brokerConfig: KafkaConfig,
                             protected val replicaManager: ReplicaManager,
                             metrics: Metrics,
                             time: Time,
-                            threadNamePrefix: Option[String] = None,
```

Comment:
You delete the `threadNamePrefix` from the constructor, but you don't update `ReplicaManagerBuilder` correspondingly. Hence, the build fails.
https://github.com/apache/kafka/blob/64aebb5621bd27ee9f2c4ab4506f32dcff267963/core/src/main/java/kafka/server/builders/ReplicaManagerBuilder.java#L112-L134